hpobench/util/data_manager.py

```diff
@@ -980,3 +981,36 @@ def _load_json(path):
         with open(path, "r") as f:
             data = json.load(f)
         return data
+
+
+class YAHPODataManager(DataManager):
+    def __init__(self, data_dir: Union[Path, str, None]):
+        if data_dir is None:
+            data_dir = hpobench.config_file.data_dir / "yahpo_data"
+        self.data_dir = data_dir
+
+        super(YAHPODataManager, self).__init__()
+
+    @lockutils.synchronized('not_thread_process_safe', external=True,
+                            lock_path=f'{hpobench.config_file.cache_dir}/lock_yahpo_raw', delay=0.5)
+    def _try_download(self):
+        """Clone the data repository."""
+        if not self.data_dir.exists():
+            # Create the data directory if not existing
+            self.create_save_directory(self.data_dir.parent)
+
+            import git
+            git.Repo.clone_from(url='https://github.com/pfistfl/yahpo_data.git',
+                                to_path=str(self.data_dir), branch='main', multi_options=['--depth 1'])
+            self.logger.info(f'Successfully cloned data from repo to {self.data_dir}')
```

Comment:
Given that `yahpo_data` is licensed under [AGPL](https://github.com/slds-lmu/yahpo_data/blob/main/LICENSE), how do you judge the compatibility with this repo's more permissive Apache-2.0 license, @KEggensperger and @PhMueller?

Reply:
And the maintainers over there just [relicensed to Apache-2.0](https://github.com/slds-lmu/yahpo_data/commit/f9b8cd24fd8e7d9cc8fb7372f9640128941255e5). My question is resolved then, I think :relaxed: